### PR TITLE
feat: Create PayButton API Route integrated with MySQL database

### PR DESCRIPTION
**Pull request by alcipir**
Description: 
- Adds a makefile, update REAMDE, fix docker-compose, add sequelize (to connect to mysql through NextJS) and required config, including initial models and migrations. 
- Adds integration with MySQL database for creating paybuttons through the API Route /api/paybutton

Test Plan: run `yarn` to update dependencies,  `make dev` to run the application and mysql, `make init-db` to run migrations and seeds. Every step should be successful.
Open up localhost:3000, then login and click on `Add PayButton`. On the browser console, should show a 201 response (successfully created)

### CODE REVIEW

**Comment by Gattaca1** [comment]
As David suggested, moving this comment from Telegram to here
Screenshot from 2022-04-09 21-59-43

are you using `sudo` to run make?
i have a permission error i haven't bothered to track down yet when running `make dev`, but `sudo make dev` fixes it
theres a lot of different things happening with this branch, i'm not sure which change is the issue

**Comment by alcipir** [comment]
Updates: 

- Removed init-db step, now `make dev` should build and run the scripts necessary to have the paybutton server working
- Added instructions on how to get testing credentials to add to .env.local

**Comment by Gattaca1** [comment]
Could you add testing credentials from supertoken to .env?

Also I'm still getting an error here 

Screenshot from 2022-04-20 22-32-12

**Comment by alcipir** [comment]
Just pushed a fix that only show social logins as long as you provide some credentials on .env file. (So it shouldn't break your app if you don't have it) 

Let me know of any more issues, and if needed testing credentials, hit me up on telegram @Gattaca1

File: .env [hunk 1]
```diff
@@ -1 +1,2 @@
-GRPC_NODE_URL=bchd.greyh.at:8335
\ No newline at end of file
+GRPC_NODE_URL=bchd.greyh.at:8335
```
Comment (Klakurka) [inline .env:1]: Where did this URL come from? We have our own BCHD node.

Comment (Klakurka) [inline .env:1]: Where did this URL come from? We have our own BCHD node.

File: README.md [hunk 2]
```diff
@@ -4,7 +4,24 @@ https://paybutton.org
...
+FACEBOOK_CLIENT_ID=<insert credentials here>
+FACEBOOK_CLIENT_SECRET=<insert credentials here>
+GITHUB_CLIENT_ID=<insert credentials here>
+GITHUB_CLIENT_SECRET=<insert credentials here>
+GOOGLE_CLIENT_SECRET=<insert credentials here>
+SUPERTOKENS_API_KEY=<insert credentials here>
+SUPERTOKENS_CONNECTION_URI=<insert credentials here>
+```
+- **If you run docker as root user, run the following commands with `sudo`** 
+- Run the following make command to build/pull the relevant docker images and run the server locally:
```
Comment (Klakurka) [inline README.md:24]: Doesn't look related to what the branch is supposed to be so should be committed separately.

Comment (Klakurka) [inline README.md:24]: Doesn't look related to what the branch is supposed to be so should be committed separately.

File: config/frontendConfig.tsx [hunk 3]
```diff
@@ -3,6 +3,20 @@ import SessionReact from 'supertokens-auth-react/recipe/session'
...
+      'facebook': process.env.FACEBOOK_CLIENT_ID && ThirdPartyEmailPasswordReact.Facebook.init(),
+      'apple': process.env.APPLE_CLIENT_ID && ThirdPartyEmailPasswordReact.Apple.init(),
+    }
+    const availableSocialProvidersNodes = Object.keys(socialProviderNodes).map(providerKey => {
+      if (socialProviderNodes[providerKey] !== undefined)
+        return socialProviderNodes[providerKey]
+    })
+    return availableSocialProvidersNodes.filter(provider => provider !== undefined)
+  }
+
```
Comment (Klakurka) [inline config/frontendConfig.tsx:19]: Goes for all the social login stuff (should be in a different branch).

Comment (Klakurka) [inline config/frontendConfig.tsx:19]: Goes for all the social login stuff (should be in a different branch).

File: docker-compose.yml [hunk 4]
```diff
@@ -1,21 +1,24 @@
 version: "3.9"
 services:
   paybutton:
+    depends_on: 
+      - mysql
```
Comment (Klakurka) [inline docker-compose.yml:5]: Adding in MySQL should have been its own branch, separated from the "add a paybutton" stuff.

https://ourmachinery.com/post/step-by-step-programming-incrementally/

Comment (Klakurka) [inline docker-compose.yml:5]: Adding in MySQL should have been its own branch, separated from the "add a paybutton" stuff.

https://ourmachinery.com/post/step-by-step-programming-incrementally/

**Review by Klakurka — CHANGES_REQUESTED** [review]

**Review by Klakurka — CHANGES_REQUESTED** [review]

**Comment by Gattaca1** [comment]
There are a ton of features in this branch, lets break it down some so its easier to deconstruct

Updating the README can happen in any branch to reflect changes, but just about everything else should have its own branch

- Adds a makefile & update REAMDE, 

- fix docker-compose, 
- Dependency changes,
- Add MySQL,

- add sequelize (to connect to mysql through NextJS) and required config, including initial models and migrations.

- Adds integration with MySQL database for creating paybuttons through the API Route /api/paybutton,
- social login changes,

**Pull request closed without merging**